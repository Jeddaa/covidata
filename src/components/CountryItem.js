import { useParams } from 'react-router-dom';
import { useSelector } from 'react-redux';
import React, { useState, useEffect } from 'react';

const CountryItem = () => {
  const { country } = useParams();
  const { rawData } = useSelector((store) => store.data);
  const filtered = rawData.filter((item) => item.Country_Region
  === country);
  const newCase = filtered.reduce((total, item) => total + parseInt(item.Confirmed, 10), 0);

  const [imageSrc, setImageSrc] = useState(null);

  const loadImage = async () => {
    try {
      const imageModule = await import(`../images/${country}.png`);
      setImageSrc(imageModule.default);
    } catch (error) {
      const imageModule = await import('../images/default.png');
      setImageSrc(imageModule.default);
    }
  };

  useEffect(() => {
    loadImage();
  }, [country]);

  return (
    <div className="flex flex-col mb-4 bg-blue-900">
      <div className="relative flex justify-between items-end">
        <img src={imageSrc} alt={country} className="p-1 sm:pl-5 w-20 sm:w-60 h-20 sm:h-48" />
        <div className="absolute inset-0 bg-blue-800 opacity-50 pr-1" />
        <h2 className="text-white z-0 text-xl text-end font-bold mb-2 px-2 pt-2">
          {country}
          :
          {` ${newCase} Cases`}
        </h2>
      </div>

      <div className="">
        {rawData
          .filter((item) => item.Country_Region === country)
          .map((item) => (
            <div key={item.Lat} className="">
              <div className="border-solid border-2 border-blue-800 text-white px-2">
                {item.Province_State && (
                  <h4 className="mt-1 font-bold">
                    Province state:
                    {` ${item.Province_State}`}
                  </h4>
                )}

                <p>
                  Confirmed Cases:
                  {` ${item.Confirmed}`}
                </p>
                <p>
                  Confirmed Deaths:
                  {` ${item.Deaths}`}
                </p>
                <p>
                  Case Fatality Ratio:
                  {` ${Number(item.Case_Fatality_Ratio).toFixed(2)}`}
                </p>
                <p>
                  Last Updated:
                  {` ${item.Last_Update}`}
                </p>
              </div>
            </div>
          ))}
      </div>
    </div>
  );
};

export default CountryItem;
