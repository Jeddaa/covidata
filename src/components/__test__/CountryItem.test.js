import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../../redux/store';
import CountryItem from '../CountryItem';

jest.mock('axios', () => ({
  get: jest.fn(),
}));

describe('App', () => {
  it('renders the home page correctly', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Provider store={store}>
            <CountryItem />
          </Provider>
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
