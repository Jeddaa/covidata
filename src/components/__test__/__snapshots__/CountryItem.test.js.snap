// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders the home page correctly 1`] = `
<div
  className="flex flex-col mb-4 bg-neutral-400"
>
  <h2
    className="text-black text-xl font-bold mb-2 px-2 pt-2"
  >
    :
     0 Cases
  </h2>
  <div
    className=""
  />
</div>
`;
